src/structures/GuildChannel.js

```diff
@@ -571,7 +571,10 @@ class GuildChannel extends Channel {
    * @readonly
    */
   get deletable() {
-    return this.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_CHANNELS, false);
+    return (
+      this.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_CHANNELS, false) &&
+      (this.guild.rulesChannelID !== this.id || this.guild.publicUpdatesChannelID !== this.id)
```

Comment:
```suggestion
      ![this.guild.rulesChannelID, this.guild.publicUpdatesChannelID].includes(this.id))
```

Reply:
Why this reaction, @NotSugden ?

Reply:
It's not exactly more readable per-se, array includes reads better when we want to check more than 2 values. Nevertheless, I don't think we do checks this way in this repository, besides, `Array#includes` has the disadvantage that it doesn't short-circuit, nor it does look consistent when followed with other conditions (`&&`, `||`, or `??`).

Reply:
It's in case there are more values, after all, it's possible.
> Array#includes has the disadvantage that it doesn't short-circuit, nor it does look consistent when followed with other conditions (&&, ||, or ??).
Yes, i know.

Reply:
> It's in case there are more values, after all, it's possible.

There aren't, and if more are added, we'll refactor then. There's no point now, though, because we know there are only 2.